### PR TITLE
chore: Update dependencies

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
Updates dependencies on tooling and ksm.

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by dgrisonnet — APPROVED** [review]
/lgtm
/approve

**Comment by CatherineF-dev** [comment]
/lgtm
/hold
Feel free to unhold when it's ready

File: .github/workflows/govulncheck.yml [hunk 1]
```diff
@@ -23,4 +23,4 @@ jobs:
           go install golang.org/x/vuln/cmd/govulncheck@latest
       - name: Run security checks
         run: |
-          govulncheck -v ./...
+          govulncheck ./...
```
Comment (rexagod) [inline .github/workflows/govulncheck.yml:26]: > The -v flag causes govulncheck to output more information when run on source. It has no effect when run on a binary.

We may lose insight into codepaths that are affected by reducing verbosity.

Comment (mrueg) [reply .github/workflows/govulncheck.yml:26]: https://github.com/kubernetes/kube-state-metrics/actions/runs/5502704010/jobs/10027175265 it looks like `-v` got removed from recent versions of govulncheck

Comment (rexagod) [reply .github/workflows/govulncheck.yml:26]: Ah! Makes sense.

**Pull request merged**